server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java

```diff
@@ -110,7 +110,7 @@ public AgentControlAnswer processControlCommand(long agentId, AgentControlComman
     @Override
     public void processHostAdded(long hostId) {
         HostVO host = hostDao.findById(hostId);
-        if (host != null) {
+        if (host != null && HypervisorType.KVM.equals(host.getHypervisorType())) {
```

Comment:
I think LXC should also be checked here (or do like a item in array check) @mlsorensen

Reply:
Looking into this a bit further @rohityadavcloud - I'm not sure why CustomServerDiscoverer extends LibvirtServerDiscoverer, or if it's even necessary for the CustomServerDiscoverer to exist since presumably custom hypervisor would provide their own Discoverer.

If not for this, I could just make the check `getHypervisorType().equals(host.getHypervisorType())`